keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/ResourceHandler.kt

```diff
@@ -52,16 +60,36 @@ interface ResourceHandler<S : ResourceSpec, R : Any> : SpinnakerExtensionPoint {
   val eventPublisher: EventPublisher?
     get() = null
 
+  private val asyncExecutor: Executor
+    get() = Executors.newCachedThreadPool()
+
   /**
    * Resolve and convert the resource spec into the type that represents the diff-able desired
    * state.
    *
    * The value returned by this method is used as the basis of the diff (with the result of
    * [current] in order to decide whether to call [create]/[update]/[upsert].
    *
+   * Note: the default implementation of this method delegates to [desiredAsync] for compatibility
+   * with Java-based plugins. Kotlin-based implementations should always override this method.
+   *
    * @param resource the resource as persisted in the Keel database.
    */
-  suspend fun desired(resource: Resource<S>): R
+  @JvmDefault
+  suspend fun desired(resource: Resource<S>): R =
+    // desiredAsync(resource, coroutineContext[CoroutineDispatcher]!!.asExecutor()).get()
```

Comment:
@robfletcher My initial attempt was to follow your suggestion and use the same dispatcher as the coroutine scope from within which this suspend function is called, but I ran into issues when the plugin blocked while executing in this context (unrelated coroutines got cancelled because the shared thread got interrupted, apparently). I then switched to a simple thread pool-backed executor below and things work as expected.